clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/HonoKafkaConsumer.java

```diff
@@ -264,15 +267,70 @@ public void setKafkaConsumerSupplier(final Supplier<Consumer<String, Buffer>> su
         kafkaConsumerSupplier = supplier;
     }
 
+    /**
+     * Suspends fetching records from all partitions assigned to this consumer (if not already suspended).
+     * That means the next poll invocations won't return any records.
+     * For already fetched records, the record handler will still be invoked.
+     *
+     * @return {@code true} if the record fetching was active before and is now getting paused.
+     */
+    public final boolean pauseRecordFetching() {
```

Comment:
As an explanation here:
This method is the way to pause the consumer without pausing the poll loop, so that rebalances can still happen and the consumer doesn't leave the active consumer group.
This approach isn't exclusively used here since it isn't very fine grained. After calling `resumeRecordFetching()` a whole batch of records (per default up to 500 records) will be received with the next poll.
In order to flatten the line of records being currently in processing, a combination with calling the `pauseRecordHandlingAndPolling(timeout)` method is used. The latter method can pause on a per-record basis (implemented in the vert.x KafkaConsumer).